### ISSUE TITLE
Flood has slow response time when rtorrent contains many torrents

### ISSUE THREAD

**Issue by Ketchup901** [Flood-UI/flood#766]
## Your Environment
* Version used:
  + commit 8d6e88c7419b13bc6c4d5a632cd92ff15f268a46
* Environment name and version:
  + Node.js version v11.10.1
  + npm version 6.8.0
  + Web browser Firefox 65.0.2
* Operating System and version: Arch Linux (kernel 4.20.13)

## Summary
Basically, flood is super slow.

This can be easily seen while adding a torrent. It can take upwards of 20-30 seconds from adding the torrent to it appearing in the torrent list. When starting a torrent, it takes the same amount of time to see the change reflected in the ui.

## Expected Behavior
It should load in 2 seconds, as the polling rate is set to.

## Steps to Reproduce
1. Do any of the following actions: starting/stopping a torrent, adding/removing a torrent
2. It will take upwards of 30 seconds for flood to see the change. The number of torrents probably have something to do with it. In my case it's about 3800, which is more than the popular webui ruTorrent can handle.

---

## Original message

Basically, flood is super slow.

This can be easily seen while adding a torrent. It can take upwards of 20-30 seconds from adding the torrent to it appearing in the torrent list. When starting a torrent, it takes the same amount of time to see the change reflected in the ui.

I have ~3800 torrents loaded so that's probably a reason, just asking if this is expected. `torrentClientPollInterval` is set to the default value (`1000 * 2`).

**Comment by noraj** [comment]
Hi @Ketchup901 Thanks for posting your issue, but can you please take time to edit it in order to fill the issue template.  Those information is very important for the developer else he won't be able to reproduce anything.

```markdown
Type: Bug Report

- [ ] Try to follow the update procedure described in the README and try again before opening this issue.
- [ ] Please check the [F.A.Q.](https://github.com/jfurrow/flood/wiki/F.A.Q.).
- [ ] Please check the [Troubleshooting](https://github.com/jfurrow/flood/wiki/Troubleshooting) wiki section.

## Your Environment
<!--- Include as many relevant details about the environment you experienced the bug in -->
* Version used:
  + Version (stable release) `git --no-pager tag`
  + Commit ID (development release) `git --no-pager log -1`
* Environment name and version:
  + Node.js version `node --version`
  + npm version `npm --version`
  + Web browser `name and version`
* Operating System and version:

## Summary
<!--- Provide a general summary of the issue in the Title above -->

## Expected Behavior
<!--- (Optional) Tell us what should happen -->

## Current Behavior
<!--- (Optional) Tell us what happens instead of the expected behavior -->

## Possible Solution
<!--- (Optional) suggest a fix/reason for the bug, -->
<!--- or ideas how to implement the addition or change -->

## Steps to Reproduce
<!--- Provide a link to a live example, or an unambiguous set of steps to -->
<!--- reproduce this bug. Include code to reproduce, if relevant -->
1.
2.
3.
4.

## Context
<!--- (Optional) What are you trying to accomplish? -->
```

I'm closing the issue until you fill it.

Feel free to ask to re-open it as soon as it's done.

**Comment by noraj** [comment]
PS : with the same polling time and 10 times less torrents (about 340) I have no hanging

**Comment by Ketchup901** [comment]
## Your Environment
* Version used:
  + commit 8d6e88c7419b13bc6c4d5a632cd92ff15f268a46
* Environment name and version:
  + Node.js version v11.10.1
  + npm version 6.8.0
  + Web browser Firefox 65.0.2
* Operating System and version: Arch Linux (kernel 4.20.13)

## Summary
Basically, flood is super slow.

This can be easily seen while adding a torrent. It can take upwards of 20-30 seconds from adding the torrent to it appearing in the torrent list. When starting a torrent, it takes the same amount of time to see the change reflected in the ui.

## Expected Behavior
It should load in 2 seconds, as the polling rate is set to.

## Steps to Reproduce
1. Do any of the following actions: starting/stopping a torrent, adding/removing a torrent
2. It will take upwards of 30 seconds for flood to see the change. The number of torrents probably have something to do with it. In my case it's about 3800, which is more than the popular webui ruTorrent can handle.

**Comment by Ketchup901** [comment]
@noraj Please reopen.

**Comment by noraj** [comment]
@Ketchup901 Thank you.

**Comment by ruCyberPoison** [comment]
Also the same issue here the rtorret take a eternity to start and take many time to load the torrent list

**Comment by Ketchup901** [comment]
In addition, viewing the torrent information ("files" or "tracker", etc) takes a very long time (up to 10 minutes to load, if it loads at all) as well.

**Comment by Ketchup901** [comment]
Flood has now started completely crashing my rtorrent if left running for 3 minutes or so. (related to https://github.com/jfurrow/flood/issues/320?). I haven't done anything or changed the config, just added another 200 torrents or so (it is now over 4000). I'm at a complete loss as to what to do about this.

**Comment by damolp** [comment]
@Ketchup901 this might be a duplicate of #405? What's the CPU Usage you're seeing that flood is using?
There's a ~~fix~~ temp hack in #405 which increased the performance of flood when I had over 5k torrents

**Comment by jfurrow** [comment]
Closing as a duplicate of https://github.com/Flood-UI/flood/issues/766

**Comment by Ketchup901** [comment]
> Closing as a duplicate of #766

You closed it as a duplicate of itself?

**Comment by jfurrow** [comment]
Woops, https://github.com/Flood-UI/flood/issues/405 is the issue I meant to reference
